src/bcf/header.rs

```diff
@@ -236,6 +236,33 @@ pub struct HeaderView {
     pub inner: *mut htslib::bcf_hdr_t,
 }
 
+///
+/// # Examples
+///   - Output sample count of a VCF file
+///   - Output sample names of a VCF file
+///   - Output sample index given a sample name of a VCF file.
+/// ```
+/// use crate::rust_htslib::bcf::{Reader, Read};
+/// use std::io::Read as IoRead;
+///
+/// let path = &"test/test_string.vcf";
+/// let mut bcf = Reader::from_path(path).expect("Error opening file.");
+/// let header = bcf.header();
+/// assert_eq!(header.sample_count(), 2);  // Sample count
+/// let mut s = String::new();
+/// for (i, mut x) in header.samples().into_iter().enumerate() {
+///     x.read_to_string(&mut s);  // Read sample name in to `s`
+///     println!("{}", s);  // output sample name
+/// }
+/// assert_eq!(header.sample_id(b"one").unwrap(), 0);  // Sample index wrapped in Option<usize>
+/// assert_eq!(header.sample_id(b"two").unwrap(), 1);  // Sample index wrapped in Option<usize>
+/// assert!(header.sample_id(b"non existent sample").is_none());  // Return none if not found
+///
+/// assert_eq!(header.contig_count(), 1); // Number of contig in header.
+/// // obtain the data type of an INFO field
+/// let (tag_type, tag_length) = header.info_type(b"S1").unwrap();
+/// let (fmt_type, fmt_length) = header.format_type(b"GT").unwrap();
+/// ```
```

Comment:
I really like these explanatory doctests!

And I think the optimum where to put tests in this case is a bit complicate and this might not be the right place for them. I would try to do two things:
1. Split this bigger test up a bit and have small little tests below for each of the functions defined for a `HeaderView`, so for `sample_count()`, `samples()`, `sample_id()`, etc. This might seem a bit repetative, but if in doubt you can [keep repetitive code from showing up](https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example). And where possible, you can even explain input parameters, [this page gives a nice and quick intro to function-specific doc comments](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html).
2. Move whatever is useful of this example code up to the `bcf::header` module level. So move it to the top of `src/bcf/header.rs` (after imports) and use the module level `//!` doc comment syntax ([see here for possible types of doc comments](https://doc.rust-lang.org/stable/reference/comments.html#doc-comments)). This way, they [will show up in the docs here](https://docs.rs/rust-htslib/0.32.0/rust_htslib/bcf/header/index.html), saving users from having to click on `HeaderView` or `Header` to get usable code examples and indicating on which to click for further infos on the exact functions.

If anything is unclear regarding those functions, please ask away here or over at Discord!